integrations/test-gitlab-v2/client.py

```diff
@@ -0,0 +1,83 @@
+import httpx
+from typing import List, Dict, Any, AsyncIterator
+from loguru import logger
+from port_ocean.context.ocean import ocean
+
+class GitLabClient:
+    def __init__(self, base_url: str, token: str):
+        self.base_url = base_url
+        self.token = token
+        self.headers = {"Authorization": f"Bearer {self.token}"}
+
+    async def _request(self, method: str, endpoint: str, **kwargs) -> dict:
+        url = f"{self.base_url}/{endpoint}"
+        async with httpx.AsyncClient() as client:
+            response = await client.request(method, url, headers=self.headers, **kwargs)
+            response.raise_for_status()  # Raise error for non-2xx status
+            return response.json()
```

Comment:
Port Ocean has it's own https_client helper with more functionality. Please use that one instead of creating your own

Reply:
I tried using port https_client but it was throwing errors, so because of the task timeline I stick to httpx